src/cryptonote_core/service_node_list.cpp

```diff
@@ -2379,6 +2374,10 @@ namespace service_nodes
     auto it = std::prev(history.end());
     m_state = std::move(*it);
     history.erase(it);
+
+
+    if (m_blockchain.sqlite_db()->height > height)
+      m_blockchain.sqlite_db()->blockchain_detached(height);
```

Comment:
Move this directly to the blockchain_detached hooks, check the height internally in the sqlite->blockchain_detached function